electricitymap/contrib/lib/models/event_lists.py

```diff
@@ -251,6 +251,30 @@ def merge_production_breakdowns(
             production_breakdowns.events.append(prod)
         return production_breakdowns
 
+    @staticmethod
+    def update_production_breakdowns(
+        production_breakdowns: "ProductionBreakdownList",
+        new_production_breakdowns: "ProductionBreakdownList",
+        logger: Logger,
+    ) -> "ProductionBreakdownList":
+        """Given a new batch of production breakdowns, update the existing ones."""
+        if len(new_production_breakdowns) == 0:
+            return production_breakdowns
+        if len(production_breakdowns) == 0:
+            return new_production_breakdowns
+
+        for event in new_production_breakdowns.events:
+            for i, existing_event in enumerate(production_breakdowns.events):
+                if event.datetime == existing_event.datetime:
+                    production_breakdowns.events[i] = ProductionBreakdown.update(
+                        existing_event, event
+                    )
+                    break
+            else:
```

Comment:
Isn't there an indentation issue here?

Reply:
This is using a for...else syntax so I believe the indentation is correct.
Here is a easier example:
```python
for x in range(6):
  print(x)
else:
  print("Finally finished!")
```
I'll double check that this is actually needed though.

Reply:
You just taught me something 😎

Reply:
Turns out it was still needed if the new list contains new datetimes and I just forgot to add tests for it.

So I added it back + tests for it.